src/test/test_addr.c

```diff
@@ -1169,6 +1171,8 @@ test_addr_parse(void *arg)
 
   (void)arg;
 
+  mock_hostname_resolver();
```

Comment:
I think the changes in the tests are meant to be in the final commit. But they are actually in the second commit.